### PR TITLE
Copy maps using Go 1.21's new "maps" package

### PR DESCRIPTION
**Pull request by 3JoB**
An attempted implementation of #1599 .

Also updated lint to 1.21, it seemed to have some issues before?

### CODE REVIEW

File: fsutil_new.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+//go:build go1.21
+// +build go1.21
```
Comment (alexandear) [inline fsutil_new.go:2]: This is a deprecated build constraint. Can be removed.

File: fsutil_new.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+package fasthttp
+
+import "maps"
+
+// Copy global map for fs.go
```
Comment (alexandear) [inline fsutil_new.go:8]: Let's follow Go [comment-way](https://go.dev/doc/effective_go#commentary): `fsMapCopy copies...`

File: fsutil_new.go (hunk 1 above)
Comment (alexandear) [reply fsutil_new.go:2]: Removed here #1650

**Comment by erikdubbelboer** [comment]
What is the benefit of this? It just complicates the code and increases the size. And map.Copy just does exactly the same so there is no speed difference. I think it's only useful if we don't have to support older Go versions anymore which isn't soon.

**Comment by leonklingele** [comment]
> What is the benefit of this? It just complicates the code and increases the size. And map.Copy just does exactly the same so there is no speed difference. I think it's only useful if we don't have to support older Go versions anymore which isn't soon.

True! I didn't check out the code of `maps.Copy` when opening my issue and think this can be closed now after looking at it.

**Pull request closed without merging**